tests/compat.sh

```diff
@@ -170,16 +170,24 @@ filter()
 {
   LIST="$1"
   NEW_LIST=""
+  EXCLMODE=""
+  CMP=""
 
   if is_dtls "$MODE"; then
-      EXCLMODE="$EXCLUDE"'\|RC4\|ARCFOUR'
-  else
-      EXCLMODE="$EXCLUDE"
+    EXCLMODE='RC4\|ARCFOUR'
+  fi
+
+  if [ "X" != "X$EXCLUDE" ]; then
+    EXCLMODE="$EXCLUDE\|$EXCLMODE"
```

Comment:
In my environment, when `EXCLMODE` is an empty string in the beginning (i.e. `MODE` doesn't pass the `is_dtls` test above), then `EXCLMODE` ends up terminated with `\|` after this line. Therefore, if mode is not DTLS and we set `EXCLUDE`, this code produces an empty list as such a malformed filter fails to match anything.

Reply:
Thanks for pointing that out. I think the last change I pushed fixes this issue.